src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -93,6 +97,7 @@ public string Path
                 throw new ArgumentNullException(nameof(context));
             }
 
+            _bufferSource = context.HttpContext.RequestServices.GetRequiredService<IRazorBufferSource>();
```

Comment:
Then this can't be used safely on multiple threads. Need to pass the `IRazorBufferSource` around everywhere.